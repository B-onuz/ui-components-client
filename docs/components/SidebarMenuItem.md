# SidebarMenuItem (Component)

Ã‰ utilizado para um Item de uma lista de menu, em conjunto com alguma tag de lista `ul`, `ol` ou relacionados.

## Como utilizar

```js
import SidebarMenuItem from 'ui-components/lib/components/SidebarMenuItem'
```

```jsx
import React from 'react'
import SidebarMenuItem from 'ui-components/lib/components/SidebarMenuItem'
import styled from 'styled-components'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faHome } from '@fortawesome/free-solid-svg-icons'

export default {
  title: 'SidebarMenuItem',
  component: SidebarMenuItem,
}

const List = styled.ul`
  list-style: none;
  padding: 0;
  background: ${(props) => props.theme.colors.primary};
`

export const SidebarMenuItemExample = () => (
  <List>
    <SidebarMenuItem icon={<FontAwesomeIcon icon={faHome} />}>Menu Item 1</SidebarMenuItem>
    <SidebarMenuItem icon={<FontAwesomeIcon icon={faHome} />}>Menu Item 2</SidebarMenuItem>
    <SidebarMenuItem icon={<FontAwesomeIcon icon={faHome} />}>Menu Item 3</SidebarMenuItem>
  </List>
)
```
