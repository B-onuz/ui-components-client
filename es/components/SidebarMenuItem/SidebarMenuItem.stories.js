'use strict'

Object.defineProperty(exports, '__esModule', {
  value: true,
})
exports.SidebarMenuItemExample = exports['default'] = void 0

var _react = _interopRequireDefault(require('react'))

var _SidebarMenuItem = _interopRequireDefault(require('./SidebarMenuItem'))

var _styledComponents = _interopRequireDefault(require('styled-components'))

var _reactFontawesome = require('@fortawesome/react-fontawesome')

var _freeSolidSvgIcons = require('@fortawesome/free-solid-svg-icons')

var _this = void 0,
  _jsxFileName = '/home/junior/Fontes/ui-components-client/src/components/SidebarMenuItem/SidebarMenuItem.stories.js'

function _interopRequireDefault(obj) {
  return obj && obj.__esModule ? obj : { default: obj }
}

function _templateObject() {
  var data = _taggedTemplateLiteralLoose(['\n  list-style: none;\n  padding: 0;\n  background: ', ';\n'])

  _templateObject = function _templateObject() {
    return data
  }

  return data
}

function _taggedTemplateLiteralLoose(strings, raw) {
  if (!raw) {
    raw = strings.slice(0)
  }
  strings.raw = raw
  return strings
}

var _default = {
  title: 'SidebarMenuItem',
  component: _SidebarMenuItem['default'],
}
exports['default'] = _default

var List = _styledComponents['default'].ul(_templateObject(), function (props) {
  return props.theme.colors.primary
})

var SidebarMenuItemExample = function SidebarMenuItemExample() {
  return /*#__PURE__*/ _react['default'].createElement(
    List,
    {
      __self: _this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 19,
        columnNumber: 3,
      },
    },
    /*#__PURE__*/ _react['default'].createElement(
      _SidebarMenuItem['default'],
      {
        icon: /*#__PURE__*/ _react['default'].createElement(_reactFontawesome.FontAwesomeIcon, {
          icon: _freeSolidSvgIcons.faHome,
          __self: _this,
          __source: {
            fileName: _jsxFileName,
            lineNumber: 20,
            columnNumber: 28,
          },
        }),
        __self: _this,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 20,
          columnNumber: 5,
        },
      },
      'Menu Item'
    )
  )
}

exports.SidebarMenuItemExample = SidebarMenuItemExample
