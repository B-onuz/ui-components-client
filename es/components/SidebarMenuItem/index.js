"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _SidebarMenuItem = _interopRequireDefault(require("./SidebarMenuItem"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var _default = _SidebarMenuItem["default"];
exports["default"] = _default;