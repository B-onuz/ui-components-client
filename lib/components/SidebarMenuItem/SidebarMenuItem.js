"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

exports.__esModule = true;
exports["default"] = void 0;

var _react = _interopRequireWildcard(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _this = void 0,
    _jsxFileName = "/Users/master/projects/bonuz/ui-components/src/components/SidebarMenuItem/SidebarMenuItem.js";

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function _templateObject3() {
  var data = _taggedTemplateLiteralLoose(["\n  background-color: rgba(255, 255, 255, 0.1);\n  padding: 16px 20px;\n  color: #fff;\n  display: block;\n  text-decoration: none;\n  cursor: pointer;\n  &:hover {\n    background-color: rgba(255, 255, 255, 0.2);\n  }\n"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteralLoose(["\n  margin-right: 8px;\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteralLoose(["\n  margin-top: 2px;\n  margin-bottom: 2px;\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteralLoose(strings, raw) { if (!raw) { raw = strings.slice(0); } strings.raw = raw; return strings; }

var StyledListItem = _styledComponents["default"].li(_templateObject());

var WrapperIcon = _styledComponents["default"].span(_templateObject2());

var MenuLink = _styledComponents["default"].a(_templateObject3());

var MenuItem = (0, _react.memo)(function (_ref) {
  var children = _ref.children,
      _ref$component = _ref.component,
      component = _ref$component === void 0 ? 'a' : _ref$component,
      as = _ref.as,
      icon = _ref.icon,
      rest = _objectWithoutPropertiesLoose(_ref, ["children", "component", "as", "icon"]);

  return /*#__PURE__*/_react["default"].createElement(StyledListItem, {
    role: "none",
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 26,
      columnNumber: 5
    }
  }, /*#__PURE__*/_react["default"].createElement(MenuLink, _extends({}, rest, {
    as: as || component,
    role: "menuitem",
    tabIndex: "-1",
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 27,
      columnNumber: 7
    }
  }), !!icon && /*#__PURE__*/_react["default"].createElement(WrapperIcon, {
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 28,
      columnNumber: 20
    }
  }, icon), " ", children));
});
var _default = MenuItem;
exports["default"] = _default;
module.exports = exports.default;