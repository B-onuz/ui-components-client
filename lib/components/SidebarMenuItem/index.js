"use strict";

exports.__esModule = true;
exports["default"] = void 0;

var _SidebarMenuItem = _interopRequireDefault(require("./SidebarMenuItem"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var _default = _SidebarMenuItem["default"];
exports["default"] = _default;
module.exports = exports.default;