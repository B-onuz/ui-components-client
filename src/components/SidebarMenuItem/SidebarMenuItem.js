import React, { memo } from 'react'
import styled from 'styled-components'
const StyledListItem = styled.li`
  margin-top: 2px;
  margin-bottom: 2px;
`

const WrapperIcon = styled.span`
  margin-right: 8px;
`

const MenuLink = styled.a`
  background-color: rgba(255, 255, 255, 0.1);
  padding: 16px 20px;
  color: #fff;
  display: block;
  text-decoration: none;
  cursor: pointer;
  &:hover {
    background-color: rgba(255, 255, 255, 0.2);
  }
`

const MenuItem = memo(({ children, component = 'a', as, icon, ...rest }) => {
  return (
    <StyledListItem role="none">
      <MenuLink {...rest} as={as || component} role="menuitem" tabIndex="-1">
        {!!icon && <WrapperIcon>{icon}</WrapperIcon>} {children}
      </MenuLink>
    </StyledListItem>
  )
})

export default MenuItem
