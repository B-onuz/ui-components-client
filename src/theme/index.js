const colors = {
  red: '#db2828',
  lightRed: '#ef4d5a',
  mediumRed: '#FF4F5D',
  darkRed: '#9c2543',
  fullDarkRed: '#550e33',
  orange: '#f2711c',
  yellow: '#fbbd08',
  olive: '#b5cc18',
  green: '#21BA45',
  teal: '#00b5ad',
  blue: '#2185d0',
  facebookBlue: '#3B5998',
  blueLink: '#4285F4',
  violet: '#6435c9',
  purple: '#a333c8',
  pink: '#e03997',
  brown: '#a5673f',
  grey: '#767676',
  textGrey: '#8D8D8D',
  lightGrey: '#6D6D6D',
  mediumGrey: '#838383',
  darkGrey: '#9A9A9A',
  boldGray: '#55555A',
  black: '#1b1c1d',
  white: '#fdfdfd',
  warning: '#7A4D05',
  success: '#1E561F',
  info: '#0E566C',
  danger: '#973937',
  cyan: '#1CB5AD',
  disabledGrey: '#ffffff1f',
  disabledGreyText: '#ffffff4d',
}

colors.primary = colors.lightRed
colors.secondary = colors.fullDarkRed
colors.default = '#e0e1e2'

const breakpoints = ['40em', '52em', '64em', '80em']

breakpoints.sm = breakpoints[0]
breakpoints.md = breakpoints[1]
breakpoints.lg = breakpoints[2]
breakpoints.xl = breakpoints[3]

const fontSizes = [12, 14, 16, 20, 24, 32]

export default {
  fontFamily: `'Lato', sans-serif`,
  colors,
  fontSizes,
  breakpoints,
}
