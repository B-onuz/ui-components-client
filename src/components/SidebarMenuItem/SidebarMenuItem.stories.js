import React from 'react'
import SidebarMenuItem from './SidebarMenuItem'
import styled from 'styled-components'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faHome } from '@fortawesome/free-solid-svg-icons'

export default {
  title: 'SidebarMenuItem',
  component: SidebarMenuItem,
}

const List = styled.ul`
  list-style: none;
  padding: 0;
  background: ${(props) => props.theme.colors.primary};
`

export const SidebarMenuItemExample = () => (
  <List>
    <SidebarMenuItem icon={<FontAwesomeIcon icon={faHome} />}>Menu Item</SidebarMenuItem>
  </List>
)
