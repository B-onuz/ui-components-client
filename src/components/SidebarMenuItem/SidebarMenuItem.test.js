import React from 'react'
import SidebarMenuItem from './SidebarMenuItem.js'
import renderer from 'react-test-renderer'
import { ThemeProvider } from 'styled-components'
import theme from '../../theme'
import 'jest-styled-components'

test('SidebarMenuItem changes the class when hovered', () => {
  const component = renderer.create(
    <ThemeProvider theme={theme}>
      <SidebarMenuItem />
    </ThemeProvider>
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
