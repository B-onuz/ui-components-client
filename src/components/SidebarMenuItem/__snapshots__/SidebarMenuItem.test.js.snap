// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidebarMenuItem changes the class when hovered 1`] = `
.c0 {
  margin-top: 2px;
  margin-bottom: 2px;
}

.c1 {
  background-color: rgba(255,255,255,0.1);
  padding: 16px 20px;
  color: #fff;
  display: block;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
}

.c1:hover {
  background-color: rgba(255,255,255,0.2);
}

<li
  className="c0"
  role="none"
>
  <a
    className="c1"
    role="menuitem"
    tabIndex="-1"
  >
     
  </a>
</li>
`;
